lib/src/metta/text.rs

```diff
@@ -664,6 +695,23 @@ mod tests {
         assert_eq!("Escaping sequence is not finished", node.message.unwrap());
     }
 
+    #[test]
+    fn test_parse_unicode() {
+        let mut parser = SExprParser::new("\"\\u{0123}\"");
+        assert_eq!(SyntaxNode::new_token_node(SyntaxNodeType::StringToken, 0..10, "\"\u{0123}\"".into()), parser.parse_string());
+
+        let mut parser = SExprParser::new("\"\\u{0123\"");
+        let node = parser.parse_string();
+        assert!(!node.is_complete);
+        assert_eq!("Invalid escape sequence", node.message.unwrap());
+
+        let mut parser = SExprParser::new("\"\\u{012345678\"");
+        let node = parser.parse_string();
+        assert!(!node.is_complete);
+        assert_eq!("Invalid escape sequence", node.message.unwrap());
+
```

Comment:
I would also add the following cases:
```
r#""\u0123}""#
r#""\u0123""#
r#""\u0{123}""#
r#""\u{defg}""#
r#""\u{130AC}""#
r#""\u{130ac}""#
```
The expected behaviour can be compared with Rust compiler parser.